packages/common/src/api/tan-query/events/useEventsByEntityId.ts

```diff
@@ -0,0 +1,53 @@
+import { Event as EventSDK, Id, OptionalId, decodeHashId } from '@audius/sdk'
+import { useQuery } from '@tanstack/react-query'
+
+import { useAudiusQueryContext } from '~/audius-query'
+import { Event } from '~/models/Event'
+import { ID } from '~/models/Identifiers'
+
+import { SelectableQueryOptions } from '../types'
+import { useCurrentUserId } from '../useCurrentUserId'
+
+import { getEventsByEntityIdQueryKey, EventsByEntityIdOptions } from './utils'
+
+const transformEvent = (event: EventSDK): Event => ({
+  ...event,
+  eventId: decodeHashId(event.eventId)!,
+  userId: decodeHashId(event.userId)!,
+  entityId: event.entityId ? (decodeHashId(event.entityId) ?? null) : null
+})
+
+type UseEventsByEntityIdOptions<TResult = Event[]> = SelectableQueryOptions<
+  Event[],
+  TResult
+> &
+  EventsByEntityIdOptions
+
+export const useEventsByEntityId = <TResult = Event[]>(
```

Comment:
nit: The default here is a bit redundant
```suggestion
export const useEventsByEntityId = <TResult>(
```